src/instructlab/model/download.py

```diff
@@ -200,9 +218,21 @@ def download(self):
             )
             raise click.exceptions.Exit(1)
 
-        model_name = self.repository.split("/")[-1]
-        os.makedirs(os.path.join(self.download_dest, model_name), exist_ok=True)
-        oci_dir = f"{DEFAULTS.OCI_DIR}/{model_name}"
+        # Try to isolate the path to where the model is stored via the URL and replicate that structure locally
+        # If unable to do that for whatever reason, just use the model name and create a folder out of that
+        model_sub_dirs = []
+        match = re.search("docker://(.*)", self.repository)
```

Comment:
I wasnt using it then because `_is_oci_repo` was only doing a prefix check, and i needed to actually extract the repo string after stripping the prefix out

I have now updated `_is_oci_repo` to return a bool for the check and also return the repo if the check is passed, so it is using _is_oci_repo here 👍